api/envoy/config/filter/network/thrift_proxy/v2alpha1/route.proto

```diff
@@ -31,6 +31,11 @@ message Route {
 
   // Route request to some upstream cluster.
   RouteAction route = 2 [(validate.rules).message = {required: true}];
+
+  // Strip the service prefix from the method name, if there's a prefix. For
+  // example, the method call Service:method would end up being just method.
+  // Defaults to false.
+  bool strip_service_name = 3;
```

Comment:
You are right, specially so because the reason for doing this is so that upstream doesn't see it (maybe it doesn't care this was locally multiplexed service -> egress listener).

Moving this to the RouteAction - thanks!